### ISSUE TITLE
[BUG] MPCTEMP tuning fails with thermal runaway

### ISSUE THREAD

**Issue by StevilKnevil** [MarlinFirmware/Marlin#25481]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I've configured MPCTemp but when I try tuning the the temperature keeps increasing while it's trying to measure heat loss at 200deg.

### Bug Timeline

Unknown

### Expected behavior

The heater should be driven to maintain a constant 200 deg

### Actual behavior

The PID calculation is unstable and runs away to infinity, keeping the heater on permanently.

### Steps to Reproduce

1. Run `m306 T`
2. Wait for the ambient heatloss phase

### Version of Marlin Firmware
Marlin bugfix-2.1.x

### Printer model
Ender3 Pro v4.2.2 board, E3d v6 Hotend, 30W heater

### Host Software
Octoprint

### Don't forget to include
- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.
[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/10929652/Marlin.zip)

### Additional information & file uploads
![image](https://user-images.githubusercontent.com/1076923/223778993-3924cd06-227c-46a6-9508-bb58e48e989b.png)

I've enabled debugging of the MPC power calculation, note how the modelled block temp diverges rapidly from actual (even going negative) causing the heater to be on full power. I've also added some debugging to the initial calculation (after it first reaches 200deg) hopefully to shed some light on why the divergence occurs.

[25481 MPCTEMP tuning fails with thermal runaway.log](https://github.com/MarlinFirmware/Marlin/files/10921768/25481MPCTEMP.tuning.fails.with.thermal.runaway.log)

**Comment by ellensp** [comment]
No Configs, Please attach Config files as requested
No hardware listed, is it a stock Ender 3 with a creality melzi controller or something else
Is it the stock hotend?

**Comment by StevilKnevil** [comment]
Sorry, I thought I'd added the configs when I added the log file! The printer model & board is listed, though I've added the hotend info now

**Comment by ellensp** [comment]
you say its a V4.2.2 board but your config says BOARD_BTT_SKR_MINI_E3_V2_0  a very different board.
What gives?

**Comment by StevilKnevil** [comment]
I've tried using the [alternative method](https://marlinfw.org/docs/features/model_predictive_control.html#configuration:~:text=constants%20at%20runtime.-,Manual%20Tuning,-If%20M306%20T) for calculating the variables using the temperatures that were reported in the attached log and got very different results.

The fastest rate of heat increase was at 6 seconds in: 
* `Rf=2.34 deg/s` 
* `Tf=17.04 deg`
* `tf=6s`

Plugging all that together gives a sensor responsiveness of 0.074, a block heat capacity of 0.087

That last result is most concerning as in theory `block responsiveness << sensor responsiveness`, but that is not the case from these calculations.

**Comment by StevilKnevil** [comment]
> you say its a V4.2.2 board but your config says BOARD_BTT_SKR_MINI_E3_V2_0 a very different board. What gives?

All apologies for that, I don't know what happened. I have uploaded the correct configs now and updated the original report, above.

**Comment by StevilKnevil** [comment]
I've been around the houses and done some experimentation.

Firstly I implemented the alternative process for calculating the sensor responsiveness by measuring the fastest rate of heating. Going through this process gave:
* Block Heat Capacity: 13.7822
* Sensor Responsiveness: 0.0662

Which went on to produce a very reasonable rest of the tuning process.

My original results gave a negative sensor responsiveness, which I believe was causing the thermal runaway. This caused me to go back an question the original mechanism for calculating the sensor responsiveness.

Given that the responsiveness can be calculated for any known sensor temperature at a given time (eq 15 in https://marlinfw.org/docs/features/model_predictive_control.html#model-details-&-tuning-algorithm) I tried calculating it for all the sampled points during the heating cycle as they all have known times and temperatures:

0: -1.3620
1: -0.9220
2: -0.7680
3: -0.6328
4: -0.5314
5: -0.4883
6: -0.4311
7: -0.3623

Interesting that they are all negative, but actually varying quite a lot whereas theory would suggest that they should be pretty much constant.
